src/sensai/evaluation/evaluation.py

```diff
@@ -0,0 +1,213 @@
+import logging
+import time
+from abc import ABC, abstractmethod
+from typing import Tuple, Dict, Any, Generator, Generic, TypeVar, Sequence
+
+import pandas as pd
+
+from .eval_stats.eval_stats_base import EvalStats, EvalStatsCollection
+from .eval_stats.eval_stats_classification import ClassificationEvalStats, ClassificationMetric
+from .eval_stats.eval_stats_regression import RegressionEvalStats, RegressionEvalStatsCollection, RegressionMetric
+from ..data_ingest import DataSplitter, DataSplitterFractional, InputOutputData
+from ..util.typing import PandasNamedTuple
+from ..vector_model import VectorClassificationModel, VectorModel, PredictorModel
+
+log = logging.getLogger(__name__)
+
+TModel = TypeVar("TModel", bound=VectorModel)
+TEvalStats = TypeVar("TEvalStats", bound=EvalStats)
+TEvalStatsCollection = TypeVar("TEvalStatsCollection", bound=EvalStatsCollection)
+
+
+# TODO: I'm not sure about the name and signature. Should we add a more generic model type and call this ModelEvaluator?
+class MetricsEvaluator(ABC):
```

Comment:
As originally suggested, I would prefer the name **MetricsComputation**, as the term "evaluator" is already reserved for different semantics. Or perhaps **MetricsDictProvider** or **MetricsDictComputation**?

Reply:
+1 for MetricsDictProvider